### ISSUE TITLE
[Feature/Improvement] Color code workouts

### ISSUE THREAD

**Issue by a-mabe** [a-mabe/OpenHIIT#7]
**Purpose:** As a user, I want to be able to color code workouts.

**More details:** 

<img width="498" alt="Screenshot 2023-09-01 at 1 26 53 PM" src="https://github.com/a-mabe/OpenHIIT/assets/39250511/b12b4492-bea4-4993-a01c-506dd7b02931">

Potentially add color picker: https://pub.dev/packages/flutter_material_color_picker

**Acceptance Criteria:**
- When creating (or editing) a workout, the user can select a color to label the workout.
- Color is used to set background color on the workout's list tile on the homepage and also when viewing the workout.

**Comment by alastairrmcneill** [comment]
Heya @a-mabe, do you want the colour of the workout to influence the Work = green and Rest = Red colour when running the workout?

**Comment by a-mabe** [comment]
Hi @alastairrmcneill, just when viewing the workouts on the homescreen and in the view workout screen itself, like these mockups:

<img width="498" alt="Screenshot 2023-09-01 at 1 26 53 PM" src="https://github.com/a-mabe/OpenHIIT/assets/39250511/febf1951-97f0-4563-ad91-fe6fba1e3a8c">

Likely would need to add a color picker during workout creation: https://pub.dev/packages/flutter_material_color_picker

And another DB column:

```
CREATE TABLE IF NOT EXISTS WorkoutTable(id TEXT PRIMARY KEY,
            title TEXT,
            numExercises INTEGER,
            exercises TEXT,
            ...
            workoutColor TEXT // Could save color name, hex value, etc.
```

Might also have to migrate the existing DB entries so as to not affect existing users.

**Comment by alastairrmcneill** [comment]
@a-mabe I've created the PR for this issue, let me know if there are any issues with the 2 PRs open at the same time

**Comment by alastairrmcneill** [comment]
Ah wait hold on, I forgot to ask you to assign it to me. Hope that doesn't cause any issues?

**Comment by a-mabe** [comment]
No worries @alastairrmcneill, I will assign the issue now. I put the issues in the [In Review column](https://github.com/users/a-mabe/projects/2) so they're on my todo list

**Comment by a-mabe** [comment]
Closed in #71. Thank you @alastairrmcneill for the contribution!
